### PR TITLE
Add wet/dry functionality for the 2D Multilayer-SWE

### PR DESCRIPTION
**Pull request by patrickersing**
Analogous to #38 this PR extends the `ShallowWaterMultiLayerEquations2D` with functionality to deal with wet/dry transitions.

### CODE REVIEW

**Review by andrewwinters5000 — CHANGES_REQUESTED** [review]
Nice work @patrickersing ! Overall everything looks good to go. Just a few small corrections in the comments

File: src/callbacks_stage/positivity_shallow_water_dg2d.jl [hunk 1]
```diff
@@ -88,4 +88,85 @@ function limiter_shallow_water!(u, threshold::Real, variable,
 
     return nothing
 end
+# !!! warning "Experimental code"
+#     This is an experimental feature and may change in future releases.
+function limiter_shallow_water!(u, threshold::Real, variable,
```
[the remark this answers is not in the document]
Comment (patrickersing) [reply src/callbacks_stage/positivity_shallow_water_dg2d.jl:93]: Yes, this should be changed to be consistent with the formatting. The formatting check didn't complain about this though.

**Review by andrewwinters5000 — APPROVED** [review]
LGTM!

**Comment by andrewwinters5000** [comment]
We again get a few uncovered lines that should be reported as covered by testing (similar to the discussion [here](https://github.com/trixi-framework/TrixiShallowWater.jl/pull/38#discussion_r1575766064)). We can still merge this into `main` but we can keep an eye on these strange lines...

**Comment by patrickersing** [comment]
I don't think we have to worry about these uncovered lines. Looking at the coverage reports from `Trixi.jl`, there are numerous untracked `end` lines when `@threaded`-loops are used. Since only the `end` is missing coverage, we can also be sure that everything is tested.  So I think we can go ahead and merge this.

**Pull request merged**